micrometer-core/src/main/java/io/micrometer/core/instrument/binder/okhttp3/OkHttpRemoveHostTagFilter.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright 2017 Pivotal Software, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.core.instrument.binder.okhttp3;
+
+import io.micrometer.core.instrument.Meter;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.config.MeterFilter;
+import org.apache.http.protocol.HttpContext;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Removes the original OkHttp host tag, leaving only the new tags (target.schema, target.host, target.port) - aligned with tags naming of Apache HttpClient
+ *
+ * @author Jakub Marchwicki
+ * @see io.micrometer.core.instrument.binder.httpcomponents.MicrometerHttpRequestExecutor#generateTagsForRoute(HttpContext)
+ */
+public class OkHttpRemoveHostTagFilter implements MeterFilter {
+
+    @Override
+    public Meter.Id map(Meter.Id id) {
+        List<Tag> newTags = id.getTags().stream()
+                .filter(t -> !t.getKey().equals("host")).collect(Collectors.toList());
```

Comment:
If a user had a commonTag setup of `host='myHost'` is there a risk that this would accidentally strip out the commonTag `host` tag?

Reply:
That takes a key as a whole (as opposed to `contains()` method. So I think we are good. 
But to be explicite I'll include a custome `myHost` tag

Reply:
if a user had custom tag `host='myHost'` that would be overwritten by the `OkHttpMetricsEventListener` anyway. That was the exact scenario I was in, where we had some common tags that were overwritten by the metrics listener. In the spring configuration, we had a customizer, which value was overwritten

```java
MeterRegistryCustomizer<MeterRegistry> meterRegistryCustomizer(Environment environment) {
    String hostName = environment.getProperty("JMX_HOSTNAME");
    return registry -> registry.config().commonTags(
        "host", hostName != null ? hostName.split("\\.")[0] : "localhost"
    );
}
```

Maybe try and replace it with a commonTag setup, if present?

Reply:
I retreat my comment ;) There is no way to retrieve common tags explicitly, which inclines me towards: maybe bluntly overwriting a common tag by `OkHttpMetricsEventListener` is a bug in a first place?

Reply:
I agree. I haven't used commonTags much myself since Prometheus has a built in concept that I leverage. However it seems like we should elevate commonTags so they always win and can't be overwritten on accident. (I think that is something we could work on in a separate issue)

Perhaps a better solution than the `OkHttpRemoveHostTagFilter` would be just to add a constructor arg in the `OkHttpMetricsEventListener` to disable it. I don't love it, but it does avoid the potential removing of a tag that was added via `commonTag` config.